ocs_ci/framework/main.py

```diff
@@ -41,10 +42,26 @@ def init_ocsci_conf(arguments=None):
     if custom_config:
         with open(os.path.expanduser(custom_config)) as file_stream:
             custom_config_data = yaml.safe_load(file_stream)
+            custom_config_data_str = str(custom_config_data)
+            if "'platform': 'vsphere'" in custom_config_data_str:
+                if not os.path.isfile(constants.VSPHERE_CONFIG_PATH):
+                    msg = "vsphere_upi_vars.yaml MUST be provided for vSphere Platform"
+                    pytest.fail(msg)
+                with open(os.path.expanduser(constants.VSPHERE_CONFIG_PATH)) as file_stream:
+                    vsphere_config_data = yaml.safe_load(file_stream)
+                    framework.config.update(vsphere_config_data)
             framework.config.update(custom_config_data)
     if cluster_config:
         with open(os.path.expanduser(cluster_config)) as file_stream:
             cluster_config_data = yaml.safe_load(file_stream)
+            cluster_config_data_str = str(cluster_config_data)
```

Comment:
This cluster_config is deprecated  since https://github.com/red-hat-storage/ocs-ci/pull/786 got merged.
If you would like to do some check and fail in case the vmware deployment we should create some separate function like validate config and in this you can do those checks once all config files are loaded. And this function can be called from the main after we load the config.

Reply:
I made the config change here how it can look like:
https://github.com/petr-balogh/ocs-ci/commit/5460b9185e6931344f2e402a99f89e7e9b541260
Or check the last commit in this branch in case I will force push later https://github.com/petr-balogh/ocs-ci/blob/vSphere-UPI/ocs_ci/framework/main.py

Reply:
cluster_config is not needed. When I started both cluster_config and ocsci_config are supported thats the reason why I am reading from both config parameters.